waku/v2/peermanager/peer_connector.go

```diff
@@ -281,6 +286,15 @@ func (c *PeerConnectionStrategy) dialPeer(pi peer.AddrInfo, sem chan struct{}) {
 		c.addConnectionBackoff(pi.ID)
 		c.host.Peerstore().(wps.WakuPeerstore).AddConnFailure(pi.ID)
 		c.logger.Warn("connecting to peer", logging.HostID("peerID", pi.ID), zap.Error(err))
+		em, emitterErr := c.host.EventBus().Emitter(new(DialError))
```

Comment:
The emitter could be part of the `PeerConnectionStrategy` struct to reuse it. That way it would not be necessary to create a new emitter every time you have a dial error. (See example in  `CreateEventEmitters()` on `topic_events.go`)

Reply:
Refactored in latest commit. The emitter is now part of `PeerConnectionStrategy` struct, created in `Start()` and closed in `Stop()`